Source/JavaScriptCore/heap/FreeList.h

```diff
@@ -94,12 +105,12 @@ class FreeList {
     unsigned cellSize() const { return m_cellSize; }
     
 private:
-    FreeCell* head() const { return FreeCell::descramble(m_scrambledHead, m_secret); }
+    FreeCell* head() const { return m_head; }
     
-    uintptr_t m_scrambledHead { 0 };
-    uintptr_t m_secret { 0 };
+    char* m_payloadStart { nullptr };
     char* m_payloadEnd { nullptr };
```

Comment:
Let's name these `m_intervalStart` and `m_intervalEnd` instead to match the algorithm.  The `payload` term is a bit ambiguous.

Reply:
Resolved in latest push.